File: src/main/java/net/cr24/primeval/util/PrimevalUtil.java
package net.cr24.primeval.util;

import java.util.List;
import net.minecraft.core.BlockPos;
import net.minecraft.tags.TagKey;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.item.ItemEntity;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;
import net.minecraft.world.phys.AABB;
import net.minecraft.world.phys.Vec3;

public class PrimevalUtil {

    @SafeVarargs
    public static boolean itemEntitiesInBlock(Level world, BlockPos pos, TagKey<Item>... tags) {
        boolean[] checks = new boolean[tags.length];
        List<Entity> itemEntities = world.getEntities((Entity) null, AABB.unitCubeFromLowerCorner(new Vec3(pos.getX(), pos.getY(), pos.getZ())), entity -> entity instanceof ItemEntity);
        for (Entity ent : itemEntities) {
            ItemEntity itemEnt = (ItemEntity) ent;
            ItemStack stack = itemEnt.getItem();
            for (int itemStep = 0; itemStep < stack.getCount(); itemStep++) {
                for (int i = 0; i < checks.length; i++) {
                    if (checks[i]) continue;
                    if (stack.is(tags[i])) {
                        checks[i] = true;
                        i = 100;
                        ent.hurt(world.damageSources().inFire(), 10);
                    }
                }
            }
        }
        for (int i = 0; i < checks.length; i++) {
            if (!checks[i]) return false;
        }
        return true;
    }
}

File: src/main/java/net/cr24/primeval/util/PrimevalUtil.kt
package net.cr24.primeval.util

import net.minecraft.entity.Entity
import net.minecraft.entity.ItemEntity
import net.minecraft.entity.damage.DamageSource
import net.minecraft.item.Item
import net.minecraft.tag.TagKey
import net.minecraft.util.math.BlockPos
import net.minecraft.util.math.Box
import net.minecraft.util.math.Vec3d
import net.minecraft.world.World

class PrimevalUtil {
    companion object {
        @SafeVarargs
        fun itemEntitiesInBlock(world: World, pos: BlockPos, vararg tags: TagKey<Item?>?): Boolean {
            val checks = BooleanArray(tags.size)
            val itemEntities = world.getOtherEntities(null, Box.from(Vec3d(pos.x.toDouble(), pos.y.toDouble(), pos.z.toDouble()))) { entity: Entity? -> entity is ItemEntity }
            println(itemEntities)
            for (ent in itemEntities) {
                val itemEnt = ent as ItemEntity
                val stack = itemEnt.stack
                for (itemStep in 0 until stack.count) {
                    var i = 0
                    while (i < checks.size) {
                        if (checks[i]) {
                            i++
                            continue
                        }
                        if (stack.isIn(tags[i])) {
                            checks[i] = true
                            i = 100
                            ent.damage(DamageSource.IN_FIRE, 10f)
                        }
                        i++
                    }
                }
            }
            for (check in checks)
                if (!check) return false
            return true
        }
    }
}